Fix PPA URL, add tools_only parameter, add start/stop checks to unit file

runejuhl opened the pull request.

runejuhl committed:
Fix PPA URL

runejuhl committed:
Add ensure parameter to control version; default to 'installed'

This makes the module a bit more safe to use as we avoid version mismatch
between the kernel module and the tools.

flokli commented on manifests/packages.pp:
```diff
@@ -1,12 +1,20 @@
+  # Wireguard tools only work with the kernel module with the same version; take
+  # care if using `latest` as it may break the tools until the module has been
+  # reloaded
+  Variant[Enum['installed', 'latest', 'present'], String] $version = 'installed',
```
I'm a bit unhappy with calling this `$version`. Maybe `$ensure` makes more sense, as it's already pretty common in some puppet resources.

runejuhl replied:
I'm fine with either. I'm not that fond of overloading `ensure` to set not just define whether a package should be present or not but also to control which version, but since it's common to use `ensure` -- let's keep it that way.

runejuhl committed:
Add parameter to avoid installation of dkms package + dependencies

runejuhl committed:
Make ExecStart and ExecStop idempotent by checking for interface

If the interface already exists when the service is started the service will
fail as `ip link add dev` and `ip link del dev` exits with code != 0.

runejuhl commented:
I've added some more minor changes to this PR -- if you want me to split it out into seperate PRs just let me know.

flokli commented on files/wireguard@.service:
```diff
@@ -11,9 +11,9 @@ Documentation=https://git.zx2c4.com/WireGuard/about/src/tools/wg.8
+ExecStartPre=/bin/sh -c 'ip link show %i >/dev/null 2>/dev/null || ip link add dev %i type wireguard'
 ExecStart=/usr/bin/wg setconf %i /etc/wireguard/%i.conf
-ExecStop=/sbin/ip link del %i
+ExecStop=/bin/sh -c 'ip link show %i >/dev/null 2>/dev/null && ip link del dev %i'
```
Hrm. I'm hoping to get rid of creating interfaces like this as soon as https://github.com/systemd/systemd/pull/4191 is ready.

In the meantime, I'd better like to use something like `ConditionPathExists=!/sys/class/net/%i`, if we want to ensure the interface doesn't yet exist. before starting the unit.

Can you rewrite all Execs to `/usr/bin/env $binary`?

runejuhl replied:
Yeah, I'm waiting for that PR to be merged myself :)

Good idea about using `ConditionPathExists`, that didn't enter my mind. I'll update it in a sec.

flokli commented on manifests/packages.pp:
```diff
@@ -1,12 +1,23 @@
+  # care if using `latest` as it may break the tools until the module has been
+  # reloaded
+  Variant[Enum['installed', 'latest', 'present'], String] $ensure = 'installed',
+  # Containers (e.g. LXC) can use wireguad but usually have no way of loading
```
typo in `wireguad`

flokli commented:
if you squash the requested changes into the existing commits, I'm fine with keeping things in the same PR :-)

runejuhl commented:
Hi @flokli -- anything left before this can be merged?
(I've made the requested changes but since I squashed them into the existing commits the timestamps weren't updated so I'm not sure if you've gotten a message that the PR changed, even though the comments are now shown as "outdated"? ...I should probably just have reset timestamp :) )

flokli commented:
@runejuhl yes, somehow I didn't receive an email about the changes. Merged now, thanks!
